File: docs/DataStructure/Binary Tree/binary_tree.cpp
#include<bits/stdc++.h>

using namespace std;

char binary_tree[10];

int root_tree(char value) {
    if (binary_tree[0] == '\0')
        binary_tree[0] = value;
    else
        cout << "Binary tree already had value of root";
}


int set_left_node(char value, int node) {
    if (binary_tree[node] == '\0')
        binary_tree[2*node+1] = value;
    else
        cout << "";
}


int set_right_node(char value, int node) {
    if (binary_tree[node] == '\0')
        binary_tree[2*node+2] = value;
    else
        cout << "";
}


int main() {
    
}

File: docs/DataStructure/Binary Tree/binary_tree.py
# Implementation of binary tree using array - Python

binary_tree = [None] * 100

def root_tree(value):
    if binary_tree[0] is None:
        binary_tree[0] = value
    else:
        print('Binary tree already had value of root')


def set_left_node(parent, value):
    if binary_tree[parent] is not None:
        binary_tree[2*parent + 1] = value
    else:
        print('Node is not exist, can not set left value of node')


def set_right_node(parent, value):
    if binary_tree[parent] is not None:
        binary_tree[2*parent+2] = value
    else:
        print('Node is not exist, can not set right value of node')


def visualize_binary_tree():
    """
         A(0)    
        /   \
        B(1)  C(2)  
    /   \      \
    D(3)  E(4)   F(6) 
    
    """

